net/core/filter.c

```diff
@@ -7981,6 +8010,71 @@ static u32 sock_ops_convert_ctx_access(enum bpf_access_type type,
 		SOCK_OPS_GET_OR_SET_FIELD(sk_txhash, sk_txhash,
 					  struct sock, type);
 		break;
+
+	case offsetof(struct bpf_sock_ops, netns_dev) ...
+	     offsetof(struct bpf_sock_ops, netns_dev) + sizeof(u64) - 1:
+#ifdef CONFIG_NET_NS
+		/* We get the netns_dev at BPF-load-time and not at
+		 * BPF-exec-time. We assume that netns_dev is a constant.
+		 */
+		res = ns_get_path_cb(&ns_path, sockops_netns_cb, NULL);
+		if (IS_ERR(res)) {
+			netns_dev = 0;
+		} else {
+			ns_inode = ns_path.dentry->d_inode;
+			netns_dev = new_encode_dev(ns_inode->i_sb->s_dev);
+		}
+		off = si->off;
```

Comment:
A nit: you could make it `off = si->off - offsetof(struct bpf_sock_ops, netns_dev);` instead of splitting it.

Reply:
I followed the same pattern elsewhere in this file:
https://github.com/kinvolk/linux/blob/11154c32377128d1b3a7939f447fa57b59360a5d/net/core/filter.c#L7239-L7240

Reply:
Alright, nothing to do here then.